static/app/components/profiling/BoundTooltip.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+import {mat3, vec2} from 'gl-matrix';
+
+import {FlamegraphTheme} from 'sentry/utils/profiling/flamegraph/FlamegraphTheme';
+import {getContext, measureText, Rect} from 'sentry/utils/profiling/gl/utils';
+import {useDevicePixelRatio} from 'sentry/utils/useDevicePixelRatio';
+
+const useCachedMeasure = (string: string): Rect => {
+  const cache = React.useRef<Record<string, Rect>>({});
+  const ctx = React.useMemo(() => {
+    const context = getContext(document.createElement('canvas'), '2d');
```

Comment:
This has the possibility of creating multiple canvases if the hook is reused. Can we ensure to always reuse the same canvas in these cases by checking if the canvas exists in the DOM first? Perhaps by giving it an id?

Reply:
Yup, 100%

Reply:
So actually, we are not appending the canvas to the DOM anyways, it just lives in memory. I think it's aight to keep it as-is, but it's a possible optimization for future 100%